File: GeneratedEntityReaderClasses/Components/GodInfoComponent.cs
namespace GeneratedNoitaClasses;
public class GodInfoComponent : NoitaComponentBase,  iNoitaType<GodInfoComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat mana_current { get; set; } = new();
	public NoitaFloat mana_max { get; set; } = new();
	public NoitaFloat gold { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		mana_current.Read(s);
		mana_max.Read(s);
		gold.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		mana_current.Write(s);
		mana_max.Write(s);
		gold.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/GodInfoComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class GodInfoComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    mana_current:NoitaFloat = field(default_factory=NoitaFloat)
    mana_max:NoitaFloat = field(default_factory=NoitaFloat)
    gold:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.mana_current.Read(s)
        self.mana_max.Read(s)
        self.gold.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.mana_current.Write(s)
        self.mana_max.Write(s)
        self.gold.Write(s)

